### PR TITLE
Test for line graph

### PR DESCRIPTION
**Pull request by nsengiyumva-wilberforce**
fix for issue https://github.com/networkx/networkx/issues/6028

### CODE REVIEW

**Comment by MridulS** [comment]
There are more red blocks at https://github.com/networkx/networkx/issues/6028 which aren't covered here, do you plan to add them in the PR? We can do another PR too :)

**Comment by nsengiyumva-wilberforce** [comment]
> There are more red blocks at #6028 which aren't covered here, do you plan to add them in the PR? We can do another PR too :)

Yeah, I am working on them. I want to finish all of them

**Comment by nsengiyumva-wilberforce** [comment]
@MridulS u can merge this, it's now at 99%, I am working on the 2 lines remaining

File: networkx/generators/tests/test_line.py [hunk 1]
```diff
@@ -275,3 +275,86 @@ def test_line_different_node_types(self):
         H = nx.line_graph(G)
         J = nx.inverse_line_graph(H)
         assert nx.is_isomorphic(G, J)
+
+    def test_line_vertex_more(self):
+        G = nx.Graph()
```
[the remark this answers is not in the document]
Comment (nsengiyumva-wilberforce) [reply networkx/generators/tests/test_line.py:280]: > Could you add a quick comment in this function? What is is supposed to test? Also it helps if the name of the test function is more informative. [do check all the other test added here too]
> 
> Thanks for your contributions! :)

Like you advised, I have renamed it to it's meaningful name, I am super grateful to take note of this in the next contribution

Thanks big @MridulS

File: networkx/generators/tests/test_line.py [hunk 2]
```diff
@@ -275,3 +275,86 @@ def test_line_different_node_types(self):
...
+        pytest.raises(nx.NetworkXError, line._odd_triangle, G, [1, 2, 3])
+
+    def test_line_starting_edge_not_in_graph(self):
+        G = nx.Graph()
+        G_edges = [(1, 2), (2, 3), (3, 1)]
+        G.add_edges_from(G_edges)
+        pytest.raises(nx.NetworkXError, line._select_starting_cell, G, (1, 4))
+
+    def test_line_rs(self):
+        G = nx.Graph()
```
Comment (MridulS) [inline networkx/generators/tests/test_line.py:334]: Thanks for all the tests @nsengiyumva-wilberforce!

Could you also add some sort of descriptions what exactly is the function supposed to test?
This description could be comment in the function. Something like:

``` python
def test_line_rs(self):
    # Test .................
    ...
    ...
```

Comment (nsengiyumva-wilberforce) [reply networkx/generators/tests/test_line.py:335]: I have commented all the methods that do not have direct names to what they are doing!


thanks big

Comment (nsengiyumva-wilberforce) [reply networkx/generators/tests/test_line.py:335]: I am going to adhere to this practice of commenting functions that do not have clear names


Glad I have learnt this, thank you very much @MridulS

**Comment by nsengiyumva-wilberforce** [comment]
hi Seth,

I am just reminding you to check on
https://github.com/networkx/networkx/pull/6108 and
https://github.com/networkx/networkx/pull/6076. I would love to complete my
outreachy application earlier.
And thanks for all the criticism on my work in this contribution period, it
has really helped me alot.

hoping to see the PRs merged or have some comments

Best Regards,
Wilberforce

File: networkx/generators/tests/test_line.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rossbar) [reply networkx/generators/tests/test_line.py:280]: I second @MridulS 's original comment - it's not immediately obvious what this is designed to test. Another thing that might help would be to use `pytest.raises` as a context manager, and use `match=` to check against the exception message. In addition to making the test more specific, it also helps improve readers' understanding of the test itself!

**Comment by MridulS** [comment]
https://app.codecov.io/gh/networkx/networkx/blob/main/networkx/generators/line.py shows 100% coverage now. Thanks for your contribution @nsengiyumva-wilberforce!

**Pull request closed without merging**